src/dune/file_tree.ml

```diff
@@ -72,11 +70,28 @@ module Dune_file = struct
           let decoder =
             Dune_project.set_parsing_context project Sub_dirs.decode
           in
-          let sub_dirs, sexps =
+          let for_subdirs =
             Dune_lang.Decoder.parse decoder Univ_map.empty
               (Dune_lang.Ast.List (Loc.none, sexps))
           in
-          Plain { path = file; sexps; sub_dirs })
+          let contents =
+            let { Sub_dirs.Dir_map.
+                  sexps ; subdir_status } = Sub_dirs.Dir_map.root for_subdirs in
+            let (sexps, subdir_status) =
+              List.fold_right parents ~init:([sexps], subdir_status)
+                ~f:(fun { Sub_dirs.Dir_map. sexps ; subdir_status } (acc_sexps, acc_subdir_status) ->
```

Comment:
It's not clear to me what's the best way to merge `(dirs ..)`, `(vendored_dirs ...)` stanzas. Seems like its desirable to just be able to override. For example:

```
$ cat dune
(subdirs foo
 (data_only_dirs bar))
$ cat foo/dune
 (data_only_dirs baz)
```

I would expect the `(data_only_dirs bar)` setting to be the one active. @diml what do you think?